indexer/Driver.cc

```diff
@@ -126,8 +158,64 @@ class Driver {
     }
     this->shutdownAllWorkers();
     this->waitForAllWorkers();
-    // Send shutdown messages to all the workers.
-    // Now wait for each of them to shutdown.
+  }
+
+  FileGuard openCompilationDatabase() {
+    auto currentPath = std::filesystem::current_path();
+    auto compdbPath = currentPath.append("compile_commands.json").string();
+
+    FILE *compDbFile = std::fopen(compdbPath.c_str(), "rb");
+    if (!compDbFile) {
+      spdlog::error("failed to open compile_commands.json: {}",
+                    std::strerror(errno));
+      std::exit(1);
+    }
+    auto fileSize =
+        std::filesystem::file_size(std::filesystem::path(compdbPath));
```

Comment:
This will throw an exception; we should use the other variant with the error code instead.